pkg/podset/podset.go

```diff
@@ -170,6 +187,10 @@ func RestorePodSpec(meta *metav1.ObjectMeta, spec *corev1.PodSpec, info PodSetIn
 		spec.Tolerations = slices.Clone(info.Tolerations)
 		changed = true
 	}
+	if !slices.Equal(spec.SchedulingGates, info.SchedulingGates) {
```

Comment:
Do we need to respect the order of schedulingGate?
I'm wondering if we can ignore the order differences.